metrics/ci_integrations/test/source/buildkite/adapter/buildkite_source_client_adapter_test.dart

```diff
@@ -732,6 +742,235 @@ void main() {
+        expect(
+          () => adapter.fetchOneBuild(pipelineSlug, null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".fetchOneBuild() throws a StateError if fetching build fails",
+      () {
+        whenFetchOneBuild().thenErrorWith();
+
+        final result = adapter.fetchOneBuild(pipelineSlug, buildNumber);
+
+        expect(result, throwsStateError);
+      },
+    );
+
+    test(
+      ".fetchOneBuild() returns a build data if fetching a build succeeds",
+      () async {
+        whenFetchOneBuild().thenSuccessWith(const BuildkiteBuild());
+
+        final result = await adapter.fetchOneBuild(pipelineSlug, buildNumber);
+
+        expect(result, isNotNull);
+      },
+    );
+
+    test(
+      ".fetchOneBuild() returns a build data with the build number equal to the fetched build's number",
+      () async {
+        const build = BuildkiteBuild(number: buildNumber);
+        whenFetchOneBuild().thenSuccessWith(build);
+
+        final result = await adapter.fetchOneBuild(pipelineSlug, buildNumber);
+
+        expect(result.buildNumber, equals(buildNumber));
+      },
+    );
+
+    test(
+      ".fetchOneBuild() returns a build data with the started at date equal to the fetched build's started at date if it is not null",
+      () async {
+        final build = BuildkiteBuild(startedAt: startedAt);
+        whenFetchOneBuild().thenSuccessWith(build);
+
+        final result = await adapter.fetchOneBuild(pipelineSlug, buildNumber);
+
+        expect(result.startedAt, equals(startedAt));
+      },
+    );
+
+    test(
+      ".fetchOneBuild() returns a build data with the started at date equal to the fetched build's finished at date if the build's started at date is null",
+      () async {
+        final build = BuildkiteBuild(startedAt: null, finishedAt: finishedAt);
+        whenFetchOneBuild().thenSuccessWith(build);
+
+        final result = await adapter.fetchOneBuild(pipelineSlug, buildNumber);
+
+        expect(result.startedAt, equals(finishedAt));
+      },
+    );
+
+    test(
+      ".fetchOneBuild() returns a build data with the started at date equal to the current date time if the fetched build's started at and finished at dates are null",
+      () async {
+        const build = BuildkiteBuild(startedAt: null, finishedAt: null);
+        whenFetchOneBuild().thenSuccessWith(build);
+
+        final result = await adapter.fetchOneBuild(pipelineSlug, buildNumber);
+
+        expect(result.startedAt, isNotNull);
+      },
+    );
+
+    test(
+      ".fetchOneBuild() maps fetched builds states according to the specification",
+      () async {
+        const states = [
+          BuildkiteBuildState.passed,
+          BuildkiteBuildState.failed,
+          BuildkiteBuildState.canceled,
+          BuildkiteBuildState.notRun,
+          BuildkiteBuildState.scheduled,
+          BuildkiteBuildState.blocked,
+          BuildkiteBuildState.notRun,
+          BuildkiteBuildState.finished,
```

Comment:
Actually, the build cannot be of the `finished` state as this state is a shortcut for a set of finished states: `passed`, `failed`, `canceled`, and `blocked`.